independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/CodestartResourceLoadersBuilder.java

```diff
@@ -120,17 +122,22 @@ private static List<ResourceLoader> getCodestartResourceLoaders(String baseCodes
         if (catalog != null) {
             // Load codestarts from catalog codestart artifacts
             final List<String> catalogCodestartArtifacts = getCodestartArtifacts(catalog);
-            for (String artifactCoords : catalogCodestartArtifacts) {
-                if (codestartsArtifacts.containsKey(artifactCoords)) {
-                    continue;
+            for (String coords : catalogCodestartArtifacts) {
+                if (codestartsArtifacts.containsKey(coords)) {
+                    // Make sure it overrides the previous codestarts
+                    codestartsArtifacts.remove(coords);
                 }
-                codestartsArtifacts.put(artifactCoords, DependencyUtils.toArtifact(artifactCoords));
+                codestartsArtifacts.put(coords, DependencyUtils.toArtifact(coords));
             }
         }
 
         // Load codestarts from the given artifacts
-        for (String codestartArtifactCoords : extraCodestartsArtifactCoords) {
-            codestartsArtifacts.put(codestartArtifactCoords, DependencyUtils.toArtifact(codestartArtifactCoords));
+        for (String coords : extraCodestartsArtifactCoords) {
+            if (codestartsArtifacts.containsKey(coords)) {
+                // Make sure it overrides the previous codestarts
+                codestartsArtifacts.remove(coords);
+            }
+            codestartsArtifacts.put(coords, DependencyUtils.toArtifact(coords));
```

Comment:
Same here.

Reply:
@aloubyansky It's a LinkedHashMap, `put` is keeping the order if already present, it needs to be removed to have the order overridden when putting again.

Reply:
I see, thanks.